modules/percolator/src/main/java/org/elasticsearch/percolator/PercolateQueryBuilder.java

```diff
@@ -238,16 +185,37 @@ public PercolateQueryBuilder(String field, String documentType, String indexedDo
         this.documentSupplier = null;
     }
 
+    protected PercolateQueryBuilder(String field, Supplier<BytesReference> documentSupplier) {
+        if (field == null) {
+            throw new IllegalArgumentException("[field] is a required argument");
+        }
+        this.field = field;
+        this.documents = Collections.emptyList();
+        this.documentXContentType = null;
+        this.documentSupplier = documentSupplier;
+        indexedDocumentIndex = null;
+        indexedDocumentId = null;
+        indexedDocumentRouting = null;
+        indexedDocumentPreference = null;
+        indexedDocumentVersion = null;
+    }
+
     /**
      * Read from a stream.
      */
     PercolateQueryBuilder(StreamInput in) throws IOException {
         super(in);
         field = in.readString();
         name = in.readOptionalString();
-        documentType = in.readOptionalString();
+        if (in.getVersion().before(Version.V_8_0_0)) {
```

Comment:
For this query type, a  7.x 'typeless' query is represented by a null `documentType` and `indexedDocumentType`. So I think that here and in `doWriteTo`, we should always assume these parameters are `null` instead of `_doc`.